docs/baseband/gsb/index.rst

```diff
@@ -9,6 +9,224 @@ the `Giant Metrewave Radio Telescope (GMRT) <http://www.gmrt.ncra.tifr.res.in/>`
+`nchan` is the number of Fourier channels.  The samples per frame for both
+rawdump and phased is::
+
+    samples_per_frame = int(timespan_of_frame * sample_rate)
+
+where ``timespan_of_frame = 0.251658240`` (see above).
+
+While a single real GSB frame is 4 or 8 megabytes, the sample files have been
+altered so that each frame contains **4 or 8 kilobytes** worth of samples while
+retaining the same frame rate.  This sets the rawdump and phased sample rates
+to 1 / 1024 their typical values::
+
+    >>> rawdump_sample_rate = (10**8 / 3) * u.Hz / 1024.
+    >>> phased_sample_rate = (10**8 / 3 / 1024.) * u.Hz / 1024.
+    >>> timespan = 2**23 / (10**8 / 3.) * u.s  # same as 0.251... seconds
+
+A single payload file can be opened with `~baseband.gsb.open` in binary
+mode.  Here we must calculate ``payloadsize``, the size of one frame in bytes;
+since rawdump samples are 4 bits, ``payloadsize`` is ``samples_per_frame /
+2``::
+
+    >>> fb = gsb.open(SAMPLE_GSB_RAWDUMP, 'rb')
+    >>> payloadsize = int((rawdump_sample_rate * timespan).to(u.one)) // 2
+    >>> payload = fb.read_payload(payloadsize, nchan=1, bps=4,
+    ...                           complex_data=False)
+    >>> payload[:4].T
+    array([[ 0., -2., -2.,  0.]], dtype=float32)
+    >>> fb.close()
+
+(``payloadsize`` for phased data is the size of one frame *divided by the
+number of binary files*.  It can be found by dividing 8 megabytes by the
+number of bytes in one complete sample, then further dividing by the number
+of files.)
+
+Opening in stream mode allows timestamp and binary files to be read in
+concert to create data frames, and also wraps the low-level routines such that
+reading and writing is in units of samples, and provides access to header
+information.  For a full list of parameters that can be passed to
+`~baseband.gsb.open` in stream mode, please see its API entry.
+
+When opening a rawdump file in stream mode, we pass the timestamp file as the
+first argument, and the binary file to the ``raw`` keyword.  ``sample_rate``
+and ``samples_per_frame`` are also required.
+
+::
+
+    >>> fh_rd = gsb.open(SAMPLE_GSB_RAWDUMP_HEADER, mode='rs',
+    ...                  raw=SAMPLE_GSB_RAWDUMP,
+    ...                  sample_rate=rawdump_sample_rate,
+    ...                  samples_per_frame=int((rawdump_sample_rate *
+    ...                                         timespan).to(u.one)))
+    >>> fh_rd.header0
+    <GSBRawdumpHeader pc: 2015 04 27 18 45 00 0.000000240>
+    >>> dr = fh_rd.read()
+    >>> dr.shape
+    (81920,)
+    >>> dr[:3]
+    array([ 0., -2., -2.], dtype=float32)
+    >>> fh_rd.close()
+
+To open a phased fileset in stream mode, we package the binary files into a
+nested tuple with the format::
+
+    ((L pol stream 1, L pol stream 2), (R pol stream 1, R pol stream 2))
+
+The nested tuple is passed to ``raw``.  ``sample_rate`` and
+``samples_per_frame`` are also required.
+
+::
+
+    >>> fh_ph = gsb.open(SAMPLE_GSB_PHASED_HEADER, mode='rs',
```

Comment:
Pre-calculated does seem better, especially as you used `phased_sample_rate` as well.